src/NzbDrone.Core/DecisionEngine/Specifications/UpgradeDiskSpecification.cs

```diff
@@ -48,18 +48,19 @@ public virtual Decision IsSatisfiedBy(RemoteAlbum subject, SearchCriteriaBase se
 
                 if (!tracksMissing && trackFiles.Any())
                 {
-                    var lowestQuality = trackFiles.Select(c => c.Quality).OrderBy(c => c.Quality.Id).First();
+                    var currentQualities = trackFiles.Select(c => c.Quality).Distinct().ToList();
+                    var currentLanguages = trackFiles.Select(c => c.Language).Distinct().ToList();
 
                     if (!_upgradableSpecification.IsUpgradable(subject.Artist.QualityProfile,
                                                                subject.Artist.LanguageProfile,
-                                                               lowestQuality,
-                                                               trackFiles[0].Language,
+                                                               currentQualities,
+                                                               currentLanguages,
                                                                _preferredWordServiceCalculator.Calculate(subject.Artist, trackFiles[0].GetSceneOrFileName()),
                                                                subject.ParsedAlbumInfo.Quality,
                                                                subject.ParsedAlbumInfo.Language,
                                                                subject.PreferredWordScore))
                     {
-                        return Decision.Reject("Existing file on disk is of equal or higher preference: {0} - {1}", lowestQuality, trackFiles[0].Language);
+                        return Decision.Reject("Existing files on disk is of equal or higher preference: {0} - {1}", currentQualities, currentLanguages);
```

Comment:
This logging needs fixing - currentQualities is a list so you get something like
```
Existing files on disk is of equal or higher preference: System.Collections.Generic.List`1[NzbDrone.Core.Qualities.QualityModel] - System.Collections.Generic.List`1[NzbDrone.Core.Languages.Language]
```